### PR TITLE
Bm improve testing

### PR DESCRIPTION
**Pull request by bamandel**
Added Parameterized tests to run both Credentials and OAuth implementations of our services. Also fixed issues with the server tests not working properly.

### CODE REVIEW

File: src/main/java/com/synopsys/integration/jira/common/rest/service/PluginManagerService.java [hunk 1]
```diff
@@ -77,16 +76,15 @@ public PluginManagerService(Gson gson, JiraApiClient jiraApiClient) {
         requestBuilder.method(HttpMethod.GET);
         requestBuilder.addHeader(ACCEPT_HEADER, MEDIA_TYPE_PLUGIN);
 
-        JiraResponse jiraResponse = jiraApiClient.get(requestBuilder.build());
-        if (jiraResponse.isStatusCodeError()) {
-            if (404 == jiraResponse.getStatusCode()) {
-                return Optional.empty();
+        try {
+            PluginResponseModel pluginComponent = jiraApiClient.get(requestBuilder.build(), PluginResponseModel.class);
```
Comment (jamesrichard91) [inline src/main/java/com/synopsys/integration/jira/common/rest/service/PluginManagerService.java:80]: So the error handling has moved into the client code?

Comment (bamandel) [reply src/main/java/com/synopsys/integration/jira/common/rest/service/PluginManagerService.java:80]: I actually reverted back to the original here after changing it in my last PR. So the functionality will be the same as it previously was https://github.com/blackducksoftware/int-jira-common/commit/cd5ebf70bfddcc5166e3a7ac72b7f8ff92f098df#diff-0a4f2c5e551cd4fdcbb89df0c78ee3fd

File: src/test/java/com/synopsys/integration/jira/common/rest/oauth1a/AuthenticateOAuthTest.java [hunk 2]
```diff
@@ -59,60 +54,91 @@ public void createAuthUrl() throws Exception {
...
         assertTrue(StringUtils.isNotBlank(url.toString()));
 
-        System.out.println(authorizationToken);
-        System.out.println(authorizationUrl);
+        System.out.println("Please copy this temporary token into the authCredentials: " + authorizationToken);
+        System.out.println("Please follow this link and copy the verification code into the authCredentials: " + authorizationUrl);
+    }
+
+    // Set the two printed values here
+    private void setTemporaryTokenAndVerificationCode() throws Exception {
```
Comment (jamesrichard91) [inline src/test/java/com/synopsys/integration/jira/common/rest/oauth1a/AuthenticateOAuthTest.java:62]: Should this be moved to the top of the test class so other user's know to set these values if they want to run these tests?

File: src/test/java/com/synopsys/integration/jira/common/rest/oauth1a/AuthenticateOAuthTest.java [hunk 3]
```diff
@@ -59,60 +54,91 @@ public void createAuthUrl() throws Exception {
...
 
         assertNotNull(oAuthCredentialsResponse);
         String token = oAuthCredentialsResponse.token;
         assertTrue(StringUtils.isNotBlank(token));
-        System.out.println(token);
+        System.out.println("Set this access token in the appropriate oauth env var: " + token);
+    }
+
+    // Add the new access token here if you don't want to use the env var (Should normally add the token to env var)
+    private void setAccessTokenIfProvided() {
```
Comment (jamesrichard91) [inline src/test/java/com/synopsys/integration/jira/common/rest/oauth1a/AuthenticateOAuthTest.java:96]: Should this be moved to the top of the test class so other user's know to set these values if they want to run these tests?

**Review by jamesrichard91 — APPROVED** [review]

File: src/test/java/com/synopsys/integration/jira/common/rest/oauth1a/AuthenticateOAuthTest.java (hunk 3 above)
Comment (bamandel) [reply src/test/java/com/synopsys/integration/jira/common/rest/oauth1a/AuthenticateOAuthTest.java:96]: For both of these I left them in the order you would want to set them, sort of like a step by step procedure. How would you prefer this class be laid out? For reference https://github.com/blackducksoftware/int-jira-common/blob/dc06f594a076a71ad3ff7cfb84fd395838c9b7c6/src/test/java/com/synopsys/integration/jira/common/rest/oauth1a/AuthenticateOAuthTest.java

File: build.gradle [hunk 4]
```diff
@@ -20,9 +20,16 @@ nexusStaging {
...
+}
+
 dependencies {
     api 'com.synopsys.integration:integration-rest:5.0.0'
     api 'com.google.oauth-client:google-oauth-client:1.31.0'
 
     testImplementation 'org.mockito:mockito-core:2.18.3'
+    testImplementation 'org.junit.jupiter:junit-jupiter-api:5.3.1'
+    testImplementation 'org.junit.jupiter:junit-jupiter-params:5.4.2'
+    testRuntimeOnly 'org.junit.jupiter:junit-jupiter-engine:5.3.1'
```
Comment (gkillough) [inline build.gradle:34]: Shouldn't these be included via common-gradle-plugin?

File: src/test/java/com/synopsys/integration/jira/common/JiraTestConstants.java [hunk 5]
```diff
@@ -0,0 +1,23 @@
+package com.synopsys.integration.jira.common;
+
+public final class JiraTestConstants {
+
```
Comment (gkillough) [inline src/test/java/com/synopsys/integration/jira/common/JiraTestConstants.java:4]: Gross

**Review by gkillough — APPROVED** [review]

File: src/test/java/com/synopsys/integration/jira/common/JiraTestConstants.java (hunk 5 above)
Comment (bamandel) [reply src/test/java/com/synopsys/integration/jira/common/JiraTestConstants.java:4]: You don't like this? I could name it better but I like having all the Test Environment Variables in a single place so I know everything I would need to add if I were new to the project.

Comment (gkillough) [reply src/test/java/com/synopsys/integration/jira/common/JiraTestConstants.java:4]: I'm talking about line 4. The whitespace.

**Review by psantos1113 — APPROVED** [review]

**Pull request merged**